tests/SourceResultCallbackAdapterTest.php

```
@@ -33,12 +33,218 @@ public function testGood()
+
+    public function testGeneratorWithExceptionInCallback(): void
+    {
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage('Callback failed');
+
+        $result = new SourceResultCallbackAdapter(function () {
+            throw new \RuntimeException('Callback failed');
+        }, 1);
+
+        // Exception should be thrown when iterating
+        foreach ($result->generator() as $item) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$item'. (undefined)

(UnusedLocalVariable)